src/Symfony/Bundle/FrameworkBundle/DependencyInjection/FrameworkExtension.php

```diff
@@ -1364,6 +1368,24 @@ private function registerLockConfiguration(array $config, ContainerBuilder $cont
         }
     }
 
+    private function registerMessageConfiguration(array $config, ContainerBuilder $container, XmlFileLoader $loader)
+    {
+        $loader->load('message.xml');
+
+        $messageToSenderMapping = array();
+        foreach ($config['routing'] as $message => $messageConfiguration) {
+            $messageToSenderMapping[$message] = array_map(function (?string $service) {
+                if (is_string($service)) {
+                    return new Reference($service);
+                }
+
+                return $service;
+            }, $messageConfiguration['senders']);
+        }
+
+        $container->getDefinition('message.asynchronous.routing.sender_locator')->setArgument(0, $messageToSenderMapping);
```

Comment:
I would keep senders lazy by using a service locator here

Reply:
Updated to use service locators for senders as well 👍